termux-shared/src/main/java/com/termux/shared/file/FileUtils.java

```diff
@@ -0,0 +1,1627 @@
+     * @param targetFilePath The {@code path} TO which the symlink file will be created.
+     * @param destFilePath The {@code path} AT which the symlink file will be created.
+     * @return Returns the {@code errmsg} if path is not a symlink file, failed to create it,
+     * otherwise {@code null}.
+     */
+    public static String createSymlinkFile(@NonNull final Context context, final String targetFilePath, final String destFilePath) {
+        return createSymlinkFile(context, null, targetFilePath, destFilePath,
+            true, true, true);
+    }
+
+    /**
+     * Create a symlink file at path.
+     *
+     * This function is a wrapper for
+     * {@link #createSymlinkFile(Context, String, String, String, boolean, boolean, boolean)}.
+     *
+     * Dangling symlinks will be allowed.
+     * Symlink destination will be overwritten if it already exists but only if its a symlink.
+     *
+     * @param context The {@link Context} to get error string.
+     * @param label The optional label for the symlink file. This can optionally be {@code null}.
+     * @param targetFilePath The {@code path} TO which the symlink file will be created.
+     * @param destFilePath The {@code path} AT which the symlink file will be created.
+     * @return Returns the {@code errmsg} if path is not a symlink file, failed to create it,
+     * otherwise {@code null}.
+     */
+    public static String createSymlinkFile(@NonNull final Context context, String label, final String targetFilePath, final String destFilePath) {
+        return createSymlinkFile(context, label, targetFilePath, destFilePath,
+            true, true, true);
+    }
+
+    /**
+     * Create a symlink file at path.
+     *
+     * @param context The {@link Context} to get error string.
+     * @param label The optional label for the symlink file. This can optionally be {@code null}.
+     * @param targetFilePath The {@code path} TO which the symlink file will be created.
+     * @param destFilePath The {@code path} AT which the symlink file will be created.
+     * @param allowDangling The {@code boolean} that decides if it should be considered an
+     *                              error if source file doesn't exist.
+     * @param overwrite The {@code boolean} that decides if destination file should be overwritten if
+     *                  it already exists. If set to {@code true}, then destination file will be
+     *                  deleted before symlink is created.
+     * @param overwriteOnlyIfDestIsASymlink The {@code boolean} that decides if overwrite should
+     *                                         only be done if destination file is also a symlink.
+     * @return Returns the {@code errmsg} if path is not a symlink file, failed to create it,
+     * or validating permissions failed, otherwise {@code null}.
+     */
+    public static String createSymlinkFile(@NonNull final Context context, String label, final String targetFilePath, final String destFilePath,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.